flink-runtime/src/test/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolderTest.java

```diff
@@ -446,6 +525,27 @@ public OperatorCoordinator create(OperatorCoordinator.Context context) {
         return holder;
     }
 
+    private static class ReorderableManualExecutorService
+            extends ManuallyTriggeredScheduledExecutorService {
+
+        private boolean pendingNewRunnables;
+
+        private final Queue<Runnable> pendingRunnables = new ArrayDeque<>();
+
+        public void setPendingNewRunnables(boolean pendingNewRunnables) {
+            this.pendingNewRunnables = pendingNewRunnables;
+        }
+
+        @Override
+        public void execute(@Nonnull Runnable command) {
+            if (pendingNewRunnables) {
+                pendingRunnables.add(command);
```

Comment:
For the real case these `Runnables` would be executed finally, and here if they are executed would not change the result since we are testing the logic before they get executed. I think I could also finally execute these `Runnables` to be more consistent with the real case.